csrc/fusion_executor/executor.cpp

```diff
@@ -179,7 +179,7 @@ std::string FusionExecutor::getStructuredCode(
             << code << "\n======================================\n\n";
   }
   if (isDebugDumpEnabled(DebugDumpOption::CudaToFile) ||
-      isDebugDumpEnabled(DebugDumpOption::DebugInfo)) {
+      isOptionEnabled(EnableOption::KernelLineInfo)) {
```

Comment:
It seems to me a little unexpected that an enable option dumps a file. Do we need this?

Reply:
I agree and I'm more than happy to remove this one. tagging @zasdfgbnm to see if there's a reason that we were dumping the cuda source in the first place.

Reply:
> I agree and I'm more than happy to remove this one. tagging @zasdfgbnm to see if there's a reason that we were dumping the cuda source in the first place.

I think the original intention was indeed dumping some useful info for analyses using ncu. This can be done `NVFUSER_ENABLE=kernel_lineinfo NVFUSER_DUMP=cuda_to_file`. Now that we have (too) many options, I'd prefer each option as simple as possible.

Reply:
> It's probably there because the line info is pretty useless without the dumped kernel. If you only provide `NVFUSER_DUMP=debug_info` and not `cuda_to_file`, then `ncu-ui` will not show the source (it gives a file not found when you try to look at the source and asks where the file is).

:exploding_head: That's a real surprise... I though I've been using the lineinfo on my local machine when I copy over just the profile file without the actual cuda source.

Reply:
The discussion here covers all the information. I have nothing more to add, and I am OK with the change. Just a small request that we should document how to use these flags in our wiki page.

Reply:
good point. I'll remember to do that before merging this one.

Reply:
Updated in https://github.com/NVIDIA/Fuser/wiki/Developer-guide#profile-kernels

Reply:
FYI, you don't have to manually copy the cuda source. you can opt in with `--import-source 1` in ncu to import the source into the report, so you don't need to manually resolve that in ncu-ui.